addons/mail/static/src/core/common/thread_model.js

```diff
@@ -208,7 +208,7 @@ export class Thread extends Record {
      *
      * Content should be fetched and inserted in a controlled way.
      */
-    messages = Record.many("Message");
+    messagesListed = Record.many("Message");
```

Comment:
The problem is that we don't have a correct modelling here, the listed messages should not be stored on the thread, but on the respective views.

Now that I think about it, we might also have inconsistencies when displaying the same thread in chatter and chat window at the same time, as the 2 will "share" the same message list on the thread.

Note: `listedMessages` would sound better in English.